prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -502,11 +502,29 @@ static void efa_rdm_cq_progress(struct util_cq *cq)
 	struct efa_rdm_cq *efa_rdm_cq;
 	struct efa_ibv_cq_poll_list_entry *poll_list_entry;
 	struct efa_domain *efa_domain;
+	struct efa_rdm_ep *efa_rdm_ep;
+	struct fid_list_entry *fid_entry;
 
 	ofi_genlock_lock(&cq->ep_list_lock);
 	efa_rdm_cq = container_of(cq, struct efa_rdm_cq, util_cq);
 	efa_domain = container_of(efa_rdm_cq->util_cq.domain, struct efa_domain, util_domain);
 
+	/**
+	 * TODO: It's better to just post the initial batch of internal rx pkts during ep enable
+	 * so we don't have to iterate cq->ep_list here.
+	 * However, it is observed that doing that will hurt performance if application opens
+	 * some idle endpoints and never poll completions for them. Move these initial posts to
+	 * the first cq read call before having a long term fix.
+	 */
+	if (!efa_rdm_cq->initial_rx_to_all_eps_posted) {
+		dlist_foreach(&cq->ep_list, item) {
+			fid_entry = container_of(item, struct fid_list_entry, entry);
+			efa_rdm_ep = container_of(fid_entry->fid, struct efa_rdm_ep, base_ep.util_ep.ep_fid.fid);
+			efa_rdm_ep_post_internal_rx_pkts(efa_rdm_ep);
+		}
+		efa_rdm_cq->initial_rx_to_all_eps_posted = false;
```

Comment:
Supposed to be `true`?

Reply:
Sigh... I was modifying it back and forth and caused it. Good catch, I will add a test to cover it too

Reply:
Fixed, also added unit tests.